src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyQueueCapacityAction.java

```diff
@@ -193,33 +180,21 @@ public Builder stepSize(int stepSize) {
       return this;
     }
 
-    public Builder upperBound(int upperBound) {
```

Comment:
I think we will need those two methods. The JVM decider will try to override upperbound for caches / queues

Reply:
I have a fn to override to max and min bounds, and another to increase/decrease. I think it is too much for the decider to calculate and come up with exact values in between that it must override to. Let me know why you think otherwise.

Reply:
the benefit of exposing upper/lower bound to decider is it allows decider to tune some certain action within a sub range of the entire {lower, upper} range. Let me use JVM decider as an example.
So the fielddata cache has a range of 0.02 - 0.4 but in JVM decider, if we only want to tune the fielddata cache within 0.2 - 0.4, the decider can override the builder to do so.

Reply:
We decided to expose this setter, I'll make the required changes. Documenting our reasoning here:

There can be several reasons for high old heap usage, cache and queues cover only a subset of them. Currently however, these are the only ones we are looking at. The heap health decider bucketizes an unhealthy heap into 3 levels: `level-1 unhealthy`, `level-2 unhealthy`, and `level-3 unhealthy`; with level-3 being where heap is worst hit.

At **level-1 unhealthy** (heap usage ~60-70%), we want to downsize queues and caches, but to a soft lower bound, not to the absolute hard lower bound. This is so that we don't over-penalize the only resources we are able to tune.

At **level-3 unhealthy** (heap usage >~90%), the same argument can hold, but now we know for sure that there are several other objects contesting for heap space, which is causing the 90% heap usage. So we tune down caches and queues to as much as possible, which is the hard lower bound provided by `setDesiredCacheMaxSizeToMin()`.

As a side note, we should separately see if these soft level based lower bounds should be present within deciders or within the `*ActionConfig` objects.

Reply:
We should pick this up as a separate PR. The soft and hard bounds should be attributes of action config objects itself, so that they can have corresponding overrides in rca.conf, and we don't maintain and pass separate bound values from deciders. 

Instead of exposing the `lowerBound()` setter, we should expose a `useSoftLowerBound()` setter, with soft lower bound being read from rca.conf.